### PR TITLE
get: refactor handling of git files and git-only repos

### PR DESCRIPTION
**Pull request by efiop**
* [x] ❗ Have you followed the guidelines in the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) list?

* [x] 📖 Check this box if this PR **does not** require [documentation](https://dvc.org/doc) updates, or if it does **and** you have created a separate PR in [dvc.org](https://github.com/iterative/dvc.org) with such updates (or at least opened an issue about it in that repo). Please link below to your PR (or issue) in the [dvc.org](https://github.com/iterative/dvc.org) repo.

* [x] ❌ Have you checked DeepSource, CodeClimate, and other sanity checks below? We consider their findings recommendatory and don't expect everything to be addressed. Please review them carefully and fix those that actually improve code or fix bugs.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/repo/get.py [hunk 1]
```diff
@@ -55,29 +59,20 @@ def get(url, path, out=None, rev=None):
...
+        except (NotDvcRepoError, NoOutputInExternalRepoError):
+            # Not a DVC repository or, possibly, path is not tracked by DVC.
+            # Fall through and try to copy from git.
             pass
 
         if os.path.isabs(path):
             raise FileNotFoundError
 
-        if not repo_dir:
-            repo_dir = cached_clone(url, rev=rev)
```
Comment (efiop) [inline dvc/repo/get.py]: We don't reuse repo_dir anymore from external_repo, but:
1) Using repo_dir outside of external_repo context is dangerous, as it is out of its context. E.g. imagine if we decide to delete it at some point. 
2) It wasn't really necessary as we cache clones anyways and copying a cache is fast; And without reusing code is much more compact.

**Comment by efiop** [comment]
For the record: Using upstream branch to test recent fixes for travis and codecov.

File: dvc/repo/get.py [hunk 2]
```diff
@@ -55,29 +59,20 @@ def get(url, path, out=None, rev=None):
...
-                # Either an uncached out with absolute path or a user error
-
-                repo_dir = repo.root_dir
-
-        except NotDvcRepoError:
-            # Not a DVC repository, continue below and copy from git
+                # Non-cached output, fall through and try to copy from git.
+        except (NotDvcRepoError, NoOutputInExternalRepoError):
+            # Not a DVC repository or, possibly, path is not tracked by DVC.
+            # Fall through and try to copy from git.
```
Comment (pared) [inline dvc/repo/get.py]: :+1:

**Comment by pared** [comment]
One think @efiop, you left `repo_dir=None` before first try, I think that can be safely removed now.

**Comment by efiop** [comment]
Thanks @pared ! FIxed.

**Pull request merged**